src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1854,7 +1980,18 @@ public void run() {
 
                 // Follow refresh logic now. Authority is valid or
                 // skipped validation
-                refreshToken(callbackHandle, null, false, request, refreshItem, false);
+                final AuthenticationResult authResult;
+                try
+                {
+                    authResult = getTokenWithRefreshToken(null, false, request, refreshItem, false);
+                }
+                catch (final AuthenticationException authException)
+                {
+                    callbackHandle.onError(authException);
+                    return;
+                }
+                
+                callbackHandle.onSuccess(authResult);
```

Comment:
As I understand at this point you can have null authResult and invoke success callback